docs/madr/decisions/014-mesh-circuit-breaker-policy.md

```diff
@@ -0,0 +1,198 @@
+
+Chosen option: Create single `MeshCircuitBreaker` with configuration divided into two sections.
+
+## Positive Consequences
+
+* Having two separate policies, each with explicit configurations makes it easier to look for appropriate Envoy stats
+  and XDP configuration values.
+* By combining two abstractions, which exist in Envoy, and calling them both combined with a name of one of these is
+  confusing, especially when you are trying to debug some issue without deep understanding of internals of our service
+  mesh.
+* Default `MeshCircuitBreaker` policy is sufficient to replicate behaviour of current version of this policy - there
+  won't be a need to include a default `MeshOutlierDetector` policy
+
+## Negative Consequences
+
+* `MeshCircuitBreaker` will be a policy with very small amount of configuration available, which may introduce some
+  clutter (at some point there may be so many policies, that it will be much more difficult to understand all of them)
+
+## Solution
+
+By dividing the configuration section into two separate, logical sections: `connectionPool` and `outlierDetection` it's
+easier to differentiate underlying envoy features without the need to split policy into two separate ones.
+
+Additional changes to the original `CircuitBreaker` policy consists of:
+
+- flattening `detectors.*` options with only single (`.consecutive`) property
+- changing word `Errors` to `Failures`
+- adding `origin` suffix to the local failures detector
+
+It means `totalErrors.consecutive`, `gatewayErrors.consecutive` and `localErrors.consecutive`
+becomes: `consecutiveTotalFailures`, `consecutiveGatewayFailures` and `consecutiveLocalOriginFailures` respectively.
+
+### Current configuration
+
+Below is the sample `MeshCircuitBreaker` configuration
+
+```yaml
+conf:
+  interval: 1s
+  baseEjectionTime: 30s
+  maxEjectionPercent: 20
+  splitExternalAndLocalErrors: false
+  thresholds:
+    maxConnections: 2
+    maxPendingRequests: 2
+    maxRequests: 2
+    maxRetries: 2
+  detectors:
+    totalErrors:
+      consecutive: 20
+    gatewayErrors:
+      consecutive: 10
+    localErrors:
+      consecutive: 7
+    standardDeviation:
+      requestVolume: 10
+      minimumHosts: 5
+      factor: 1.9
+    failure:
+      requestVolume: 10
+      minimumHosts: 5
+      threshold: 85
+
+```
+
+### Specification
+
+#### Top level
+
+As a part of this MADR root `targetRef`'s kind can be configured as one
+of `Mesh|MeshSubset|MeshService|MeshServiceSubset`. When we finally introduce `MeshHTTPRoute` kind and if there will be
+a reasonable use case, in the next iteration `MeshGatewayRoute` and `MeshHTTPRoute` could be added as a viable kinds.
+
+```yaml
+targetRef:
+  kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
+  name: ...
+```
+
+#### From level
+
+`MeshCircuitBreaker` is outbound only policy, so only `to` should be configured.
+
+#### To level
+
+```yaml
+to:
+- targetRef:
+    kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
```

Comment:
```suggestion
    kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
```

To me only Mesh and MeshService is feasible here but I could be wrong

Reply:
addressed